src/storage/core.rs

```diff
@@ -712,13 +726,43 @@ where
                             .with_context(|| {
                                 anyhow!(format!("failed to save corrupted blob {:?}", file))
                             })?;
+                        corrupted += 1;
                     } else {
                         return Err(e.context(msg));
                     }
                 }
             }
         }
-        Ok(blobs)
+        Ok((blobs, corrupted))
+    }
+
+    async fn count_old_corrupted_blobs(config: &Config) -> usize {
+        let mut corrupted = 0;
+
+        if let Some(work_dir_path) = config.work_dir() {
+            let mut corrupted_dir_path = work_dir_path.to_path_buf();
+            corrupted_dir_path.push(config.corrupted_dir_name());
+            if corrupted_dir_path.exists() {
+                let dir = read_dir(&corrupted_dir_path).await;
+                if dir.is_err() {
+                    warn!("can't read corrupted blobl dir: {}", corrupted_dir_path.display());
```

Comment:
Typo: `"can't read corrupted blob dir: {}"`
Also, add error description to the message